### ISSUE TITLE
Removing the if statements in handleCommand()

### ISSUE THREAD

**Issue by Shu-AFK** [Shu-AFK/Datatransmission#34]
This can be done using this approach 
<img width="470" alt="image" src="https://github.com/Shu-AFK/Datatransmission/assets/69324179/fc7b014d-e212-4d0a-81b7-be3478795561">

**Comment by hvostov** [comment]
Hello, i tryed to run your program, but it doesn't have start script in this path (..\..\Server\src\scripts\run_exec.bat). Could you add it?

**Comment by Shu-AFK** [comment]
According to the current configuration, the run_exec.bat script should be generated automatically upon running the Server.exe file by the create_start_script() function inside Server::setupPort().

This function is attempting to create the run_exec.bat file at the specified relative path. If there's an issue with the creation of the script, it could be due to:
- The relative path defined in the function might not be correct.
- There may not be permission to create a file in the defined location.
- The directory path where the file needs to be created may not exist.

The create_start_script() function also logs an error message "Failed to create start script" into a log object if it fails to create the script for some reason. Can you please check any logs or console outputs for this specific error message? Any error message you can provide will be very helpful in troubleshooting this further.

**Comment by hvostov** [comment]
Forget to say, the build.bat script couldn't create Server.exe, it was a problem with sqlite3.dll. May be i do smth wrong, but the sqlite3.dll is downloaded from official site and put in the root. 

In command line i get this error "C:\Users\ahvostov\sqlite-dll-win-x64-3450300\sqlite3.dll : fatal error LNK1107: invalid or corrupt file: cannot read at 0x370 [C:\Users\ahvostov\Datatransmission\build\Server\Server.vcxproj]".

Then i tried to do it with sqlite3.lib file and Server.exe was created, but "Failed to create start script". So problem apparantly that I shouldn't build with .lib file.

**Comment by Shu-AFK** [comment]
Hmm, with the lib file it should work fine, you could try to comment this in Server::setupPort()

```cpp
if(create_start_script(cwd.string(), port) != 0) {
        log << "Failed to create start script" << std::endl;
        return false;
    }
```

and create your own run_exec.bat file like this for now
```
start "" "[Absolute path to Server.exe]\Server.exe" "Port"
```

**Comment by hvostov** [comment]
Now it just don't logs "Failed to create start script". In console "The system cannot find the path specified." and program hangs.

**Comment by Shu-AFK** [comment]
The path should be "C:\Users\ahvostov\Datatransmission\build\Server\Release\Server.exe" instead of "\c\Users\ahvostov\Datatransmission\build\Server\Release\Server.exe"

**Comment by Shu-AFK** [comment]
Does it still say that it cant find the path?

**Comment by hvostov** [comment]
Yes, the same msg "The system cannot find the path specified."

**Comment by Shu-AFK** [comment]
Have you tried to go to the executable in file explorer and copy the absolute path from there?

**Comment by Shu-AFK** [comment]
try to run the program without --set-startup

**Comment by hvostov** [comment]
without this it said : 
C:/Users/ahvostov/Datatransmission/build/Server/Release/Server.exe: error while loading shared libraries: lz4.dll: cannot open shared object file: No such file or directory

I put it near the Server.exe, and again : 
"The system cannot find the path specified."

**Comment by hvostov** [comment]
yes

**Comment by Shu-AFK** [comment]
Since I didn't get a .lib file i used the .dll to link with and it works fine on my machine, but typically it should be a .lib file yeah

**Comment by Shu-AFK** [comment]
Have you added the vcpkg bin directory to your path?

**Comment by hvostov** [comment]
the same problem...

$ ./build.bat "C:\Users\ZBook\vcpkg\installed\x64-windows\bin" "C:\Users\ZBook\vcpkg"

sqlite3.dll, lz4.dll placed in bin folder. 

Got error: C:\Users\ZBook\vcpkg\installed\x64-windows\bin\sqlite3.dll : fatal error LNK1107: invalid or corrupt file: cannot read at
 0x308 [C:\Users\ZBook\Datatransmission\build\Server\Server.vcxproj]

**Comment by Shu-AFK** [comment]
This is really weird, I'll check tomorrow when I have time. I'm not really sure why that is though

**Comment by hvostov** [comment]
Instead of setting the direct path, i wrote this in CMakeLists:
```
find_package (SQLite3)
include_directories(${SQLite3_INCLUDE_DIRS})
target_link_libraries (Server ${SQLite3_LIBRARIES})
```
With this it builds succesfully. Than i dealed with automatically creating of run_exec.bat:
`std::ofstream run_script(R"(..\..\Server\src\scripts\run_exec.bat)");  ----> std::ofstream run_script(R"(..\..\..\Server\src\scripts\run_exec.bat)");  `

**Comment by Shu-AFK** [comment]
Yeah, that's the right behaviour, the issue was with the different ide build systems. I'm glad it works now.

**Comment by hvostov** [comment]
Now client don't see lz4.dll, i put it near him and it works. When i try to connect to server client says "Credentials are invalid" and server stops with msg "The system cannot find the path specified.". The name and password are the same. Did you have this problem?
My commands for server:
`./Server.exe -p 27015 -n alexey qwe123 --set-startup`
for client: 
`./Client.exe -s 127.0.0.1 -p 27015 -u alexey -w qwe123`

**Comment by Shu-AFK** [comment]
So everything works fine now?

**Comment by hvostov** [comment]
Yes, everything works fine. But I guess we should deal with cmake building
for other potential developers.
Can I write you here, on email ?

чт, 18 апр. 2024 г., 18:27 Floyd ***@***.***>:

**Comment by Shu-AFK** [comment]
Do you have discord? I feel like thats easier

**Comment by hvostov** [comment]
I haven't,  but ok,  I'll install tomorrow

чт, 18 апр. 2024 г., 23:27 Floyd ***@***.***>:

**Comment by hvostov** [comment]
my nickname ahvostov

**Comment by Shu-AFK** [comment]
I added you
